packages/expo-modules-core/ios/Swift/Logging/PersistentFileLog.swift

```diff
@@ -0,0 +1,149 @@
+    PersistentFileLog.serialQueue.async {
+      self._ensureFileExists()
+      let size = self._getFileSize()
+      if size == 0 {
+        self._appendTextToFile(text: entry)
+      } else {
+        self._appendTextToFile(text: "\n" + entry)
+      }
+      completionHandler?(nil)
+    }
+  }
+
+  /**
+   Filter existing entries and remove ones where filter(entry) == false
+   */
+  public func filterEntries(filter: @escaping PersistentFileLogFilter, _ completionHandler: @escaping PersistentFileLogCompletionHandler) {
+    PersistentFileLog.serialQueue.async {
+      self._ensureFileExists()
+      do {
+        let contents = try self._readFileSync()
+        let newcontents = contents.filter { entry in filter(entry) }
+        try self._writeFileSync(newcontents)
+        completionHandler(nil)
+      } catch {
+        completionHandler(error)
+      }
+    }
+  }
+
+  /**
+   Clean up (remove) the log file
+   */
+  public func clearEntries(_ completionHandler: @escaping PersistentFileLogCompletionHandler) {
+    PersistentFileLog.serialQueue.async {
+      do {
+        try self._deleteFileSync()
+        completionHandler(nil)
+      } catch {
+        completionHandler(error)
+      }
+    }
+  }
+
+  // MARK: - Private methods
+
+  private func _ensureFileExists() {
+    if !FileManager.default.fileExists(atPath: filePath) {
+      FileManager.default.createFile(atPath: filePath, contents: nil)
+    }
+  }
+
+  private func _getFileSize() -> Int {
+    // Gets the file size, or returns 0 if the file does not exist
+    do {
+      let attrs: [FileAttributeKey: Any?] = try FileManager.default.attributesOfItem(atPath: filePath)
+      return attrs[FileAttributeKey.size] as? Int ?? 0
+    } catch {
+      return 0
+    }
+  }
+
+  private func _appendTextToFile(text: String) {
+    if let data = text.data(using: .utf8) {
+      if let fileHandle = FileHandle(forWritingAtPath: filePath) {
+        fileHandle.seekToEndOfFile()
+        fileHandle.write(data)
+        fileHandle.closeFile()
+      }
+    }
+  }
+
+  private func _readFileSync() throws -> [String] {
+    return try _stringToList(String(contentsOfFile: filePath, encoding: .utf8))
+  }
+
+  private func _writeFileSync(_ contents: [String]) throws {
+    if contents.isEmpty {
+      try _deleteFileSync()
+      return
+    }
+    try contents.joined(separator: "\n").write(toFile: filePath, atomically: true, encoding: .utf8)
+  }
+
+  private func _deleteFileSync() throws {
+    if FileManager.default.fileExists(atPath: filePath) {
+      try FileManager.default.removeItem(atPath: filePath)
+    }
+  }
+
+  private func _stringToList(_ contents: String?) -> [String] {
+    // If null contents, or 0 length contents, return empty list
+    return (contents != nil && contents?.lengthOfBytes(using: .utf8) ?? 0 > 0) ?
+      contents?.components(separatedBy: "\n") ?? [] :
+      []
```

Comment:
```suggestion
    guard let contents = contents, !contents.isEmpty else {
      return []
    }
    return contents.components(separatedBy: "\n")
```
I think it is more readable.